crates/router/src/connector/braintree/braintree_graphql_transformers.rs

```diff
@@ -1384,21 +1507,47 @@ impl TryFrom<&BraintreeRouterData<&types::PaymentsCompleteAuthorizeRouterData>>
         .change_context(errors::ConnectorError::MissingConnectorRedirectionPayload {
             field_name: "three_ds_data",
         })?;
-        let query = match utils::PaymentsCompleteAuthorizeRequestData::is_auto_capture(
-            &item.router_data.request,
-        )? {
-            true => CHARGE_CREDIT_CARD_MUTATION.to_string(),
-            false => AUTHORIZE_CREDIT_CARD_MUTATION.to_string(),
+
+        let customer_acceptance = item.router_data.request.customer_acceptance.clone();
+        let (query, transaction_body) = if customer_acceptance.is_some()
+            && item.router_data.request.setup_future_usage
+                == Some(common_enums::FutureUsage::OffSession)
+        {
+            (
+                match utils::PaymentsCompleteAuthorizeRequestData::is_auto_capture(
+                    &item.router_data.request,
+                )? {
+                    true => CHARGE_AND_VAULT_TRANSACTION_MUTATION.to_string(),
+                    false => AUTHORIZE_AND_VAULT_CREDIT_CARD_MUTATION.to_string(),
+                },
+                TransactionBody::Vault(VaultTransactionBody {
+                    amount: item.amount.to_owned(),
+                    merchant_account_id: metadata.merchant_account_id,
+                    vault_payment_method_after_transacting: TransactionTiming {
+                        when: "ALWAYS".to_string(),
+                    },
+                }),
+            )
+        } else {
+            (
+                match utils::PaymentsCompleteAuthorizeRequestData::is_auto_capture(
+                    &item.router_data.request,
+                )? {
+                    true => CHARGE_CREDIT_CARD_MUTATION.to_string(),
+                    false => AUTHORIZE_CREDIT_CARD_MUTATION.to_string(),
+                },
+                TransactionBody::Regular(RegularTransactionBody {
+                    amount: item.amount.to_owned(),
+                    merchant_account_id: metadata.merchant_account_id,
+                }),
+            )
```

Comment:
This is repetitive code same as `Authorize` flow, can we form a `from` function for this block